src/formio/validators/MinMaxTimeValidator.js

```diff
@@ -40,7 +76,16 @@ const MinMaxTimeValidator = {
   check(component, setting, value) {
     if (!value) return true;
 
-    return validateTimeBoundaries(component.component.minTime, component.component.maxTime, value);
+    const {isValid, error} = validateTimeBoundaries(
+      component.component.minTime,
+      component.component.maxTime,
+      value
+    );
+
+    if (!isValid) {
+      set(component, 'openForms.validationErrorContext', {timeError: error});
+    }
```

Comment:
I'm guessing that the validator instance is only created once and you can't put this on the `this` object?

In this form, wouldn't there also be a race condition risk if multiple validators run in parallel (due to `await` syntax being used or `validateAsync` being called)? What happens if the component is part of a repeating group, does this create separate component instances or is this context shared between all of them?

Reply:
> I'm guessing that the validator instance is only created once and you can't put this on the this object?

Yes, this is why I put it in the component!

> wouldn't there also be a race condition risk if multiple validators run in parallel (due to await syntax being used or validateAsync being called)

I don't think so :thinking: For the time component there is only the minTime/maxTime validation (none of the async validators apply, and there aren't any other validators available). If we were to use the same approach for other components that have more validators, then maybe the key of the errors in the context should be based on the validator type rather than the component type (i.e. `MinMaxTimeValidatorError` instead of `timeError`?) :thinking: 

> What happens if the component is part of a repeating group, does this create separate component instances or is this context shared between all of them?

It goes in the context of the component (it is not shared by other components in the same group).
![Screenshot from 2023-09-06 10-33-39](https://github.com/open-formulieren/open-forms-sdk/assets/19154114/6b629dd9-7448-4229-a3d5-6447076b7916)